### PR TITLE
bugfix: fix the wrong rollback sequence caused by the same record request from different transaction branches on different servers

### LINKED ISSUE #2186: undo log 回滚失败，回滚数据为脏数据（Has dirty records when undo）
图片
如图所示，报异常之后，分支事物回滚，提示有脏数据：
 branchRollback failed reason [Branch session rollback failed and try again later xid = 192.168.80.79:18091:2031581034 branchId = 2031581658 Has dirty records when undo.]

并且通过测试发现不是100% 重现，有回滚成功的情况

### PR DESCRIPTION
**Pull request by slievrly**
Signed-off-by: slievrly <slievrly@163.com>

### Ⅰ. Describe what this PR did

bugfix: fix the wrong rollback sequence caused by the same record request from different transaction branches on different servers

### Ⅱ. Does this pull request fix one issue?

fix #2186

### CODE REVIEW

**Comment by feijiing** [comment]
图片
将seata server 版本升级至 1.1.0-SNAPSHOT版本，另外BRANCH_TABLE 表结构同步到最新表结构，可查看seata 源码中的sql脚本，经过测试验证能够回滚成功。
问题总结1： seata server 启动没有加 -n 参数，生产环境多机，多台TC必须加的，每台区分不同的node，用于生成不同范围的事务id，不然在开启全局事务时id会冲突报错

File: script/server/db/mysql.sql [hunk 1]
```diff
@@ -31,10 +31,11 @@ CREATE TABLE IF NOT EXISTS `branch_table`
     `status`            TINYINT,
     `client_id`         VARCHAR(64),
     `application_data`  VARCHAR(2000),
-    `gmt_create`        DATETIME,
-    `gmt_modified`      DATETIME,
+    `gmt_create`        DATETIME(6),
```
Comment (zjinlei) [inline script/server/db/mysql.sql]: Milliseconds is enough.
User upgrades may forget this, might be better to add a long type timestamp column.

Comment (slievrly) [reply script/server/db/mysql.sql]: I think the long type is less readable.

File: script/server/db/mysql.sql [hunk 2]
```diff
@@ -31,10 +31,11 @@ CREATE TABLE IF NOT EXISTS `branch_table`
...
+    `gmt_modified`      DATETIME(6),
     PRIMARY KEY (`branch_id`),
-    KEY `idx_xid` (`xid`)
+    KEY `idx_xid` (`xid`),
+    KEY `idx_gmt_create` (`gmt_create`)
```
[the remark this answers is not in the document]
Comment (slievrly) [reply script/server/db/mysql.sql]: already remove gmt_create index.

**Pull request merged**